apps/src/constants.js

```diff
@@ -65,11 +67,17 @@ exports.TestResults = {
 
   // The level was solved in an optimal way.
   FREE_PLAY: 30,   // The user is in free-play mode.
+  MINIMUM_OPTIMAL_RESULT: 30,
   PASS_WITH_EXTRA_TOP_BLOCKS: 31, // There was more than one top-level block.
   EDIT_BLOCKS: 70, // The user is creating/editing a new level.
 
   // The level was solved in the ideal manner.
-  ALL_PASS: 100
+  ALL_PASS: 100,
+
+  SUBMITTED_RESULT: 1000,
+  LOCKED_RESULT: 1001,
+  REVIEW_REJECTED_RESULT: 1500,
+  REVIEW_ACCEPTED_RESULT: 2000,
```

Comment:
They need to be duplicated because there's no easy way to share them between ruby and JS code. It is probably worth adding a comment in both places referencing the other.